### ISSUE TITLE
Recent Files sometimes don't work

### ISSUE THREAD

**Issue by karlitschek** [nextcloud/server#10770]
If you click on recent files in the left side bar you should see your recent changed files. This works on my private instance but on cloud.nc.com I only see an empty page.

**Comment by MorrisJobke** [comment]
cc @rullzer @nickvergessen @skjnldsv

**Comment by koarlmichael** [comment]
Here the same with Nextcloud 14.0.2. I only see an empty page on recent files.

**Comment by MorrisJobke** [comment]
Same here: 

This request:

/apps/files/api/v1/recent

Just returns nothing:

```json
{"files":[]}
```

This is how it should look like:

```json
{"files":[{"id":"135","parentId":"2","mtime":1539855850000,"name":"Bildschirmfoto...
```

Or it should have a decent empty screen:

![bildschirmfoto 2018-10-18 um 16 55 54](https://user-images.githubusercontent.com/245432/47163647-b8dbcb00-d2f6-11e8-8c7e-b8cd89378db5.png)

**Comment by violoncelloch** [comment]
Are there groupfolders active for the users/instances it doesn't work?
I also encountered this one a while ago, but for me it only happens for users which have access to a groupfolder. See nextcloud/groupfolders#124

**Comment by MorrisJobke** [comment]
> Are there groupfolders active for the users/instances it doesn't work?

On the instance of @karlitschek it isn't.

**Comment by karlitschek** [comment]
yep. no groupfolders here

**Comment by Schmuuu** [comment]
Hi,
I just stumbled across this issue and can confirm it happens on NC15 still. However, I noticed something strange and wonder if the root cause of this issue might be a wrong time range for "recent":
![grafik](https://user-images.githubusercontent.com/26191454/50292715-1d8be100-0472-11e9-95d9-5ff69c1a8b2c.png)

... the future.
Of course, I don't know why this file will be edited on May 12th in 2034 at 12:15 first, but it seems to be listed due to this fact ;)

**Comment by MorrisJobke** [comment]
We just looked into this and the reason for this behavior is that shared folders are not fetched for the "recent files" view:

https://github.com/nextcloud/server/blob/9d9f6903c9e2a94f2a1b230fd35ec586b18b7a77/lib/private/Files/Node/Folder.php#L364-L414

It only fetches from real storages with a storage ID that stored in the file cache, but shares are only virtual mount points into the user folder and thus it will not fetch the recent files from them properly.

cc @icewind1991 @rullzer Any clever idea how to also include those subfolders or single files of storages in the SQL query or should we inject them separately somehow?

**Comment by rullzer** [comment]
So. it seems we have code duplication here I think. because we could just use the cache searching. There shared folders work. (it is the thing we use for the dav searches as well). It is not the most effective code but good enough for now I'd say.

**Comment by guzzisti** [comment]
I have a similar issue with "recent" files, the file list is completely empty although i have numerous files uploaded/updated today.
![screenshot_2019-02-03_14-41-49](https://user-images.githubusercontent.com/10488791/52176884-7cef4780-27b9-11e9-8b3a-7b82517ddedd.png)

Running NextCloud 15.0.2 with PHP 7.2 and local storage only.

**Comment by MorrisJobke** [comment]
Fix is in #14185
